src/main/java/com/purbon/kafka/topology/Configuration.java

```diff
@@ -374,11 +378,40 @@ public String getKafkaConnectClusterId() {
         .collect(Collectors.toMap(Pair::getKey, Pair::getValue));
   }
 
-  public String getKSQLServer() {
-    return config.getString(PLATFORM_SERVER_KSQL);
+  public URL getKSQLServer() {
+    if (config.hasPath(PLATFORM_SERVER_KSQL_URL)) {
+      try {
+        return new URL(config.getString(PLATFORM_SERVER_KSQL_URL));
+      } catch (MalformedURLException e) {
+        throw new ConfigurationException(config.getString(PLATFORM_SERVER_KSQL_URL) + " is not a valid KSQL URL");
+      }
+    }
+    return null;
+  }
+
+  public BasicAuth getKSQLBasicAuth() {
+    return new BasicAuth(config.getString(PLATFORM_SERVER_KSQL_BASIC_AUTH_USER), config.getString(PLATFORM_SERVER_KSQL_BASIC_AUTH_PASSWORD));
   }
 
   public boolean hasKSQLServer() {
-    return config.hasPath(PLATFORM_SERVER_KSQL);
+    return config.hasPath(PLATFORM_SERVER_KSQL_URL);
+  }
+
+  public static class BasicAuth {
```

Comment:
might be located outside of this file? i think it could be as well used in other parts of the project such as the mds api.

Reply:
moved into dedicated class